### PR TITLE
Use remote.Head to check for remote image existence

### PR DESCRIPTION
**Pull request by ImJasonH**
This updates the go-containerregistry dep to pick up [`remote.Head`](https://godoc.org/github.com/google/go-containerregistry/pkg/v1/remote#Head), and uses it in remote's `Image.Found`.

This should avoid incurring unnecessary "pulls" for Dockerhub images especially, as HEAD requests do not count as a "pulls" for the purposes of their rate limiting.

Addresses part of https://github.com/buildpacks/pack/issues/899

### CODE REVIEW

**Comment by ImJasonH** [comment]
This small change by itself failed unit tests with:

```
Suite: Image
Total: 47 | Focused: 0 | Pending: 0
...........................s....--- FAIL: TestRemote (15.83s)
    docker_registry.go:49: run registry
    --- FAIL: TestRemote/Image (13.66s)
        --- FAIL: TestRemote/Image/#Rebase/image_exists/switches_the_base (2.04s)
.panic: Not Implemented [recovered]
	panic: Not Implemented

goroutine 356 [running]:
testing.tRunner.func1.1(0x15be300, 0x17241d0)
	/usr/local/go/src/testing/testing.go:1057 +0x30d
testing.tRunner.func1(0xc000683080)
	/usr/local/go/src/testing/testing.go:1060 +0x41a
panic(0x15be300, 0x17241d0)
	/usr/local/go/src/runtime/panic.go:969 +0x175
github.com/buildpacks/imgutil/remote.(*subImage).ConfigFile(0xc0004c3120, 0xc0000295e8, 0x40, 0xc000029501)
	/Users/jasonhall/go/src/github.com/buildpack/imgutil/remote/remote.go:541 +0x39
github.com/google/go-containerregistry/pkg/v1/mutate.Rebase(0x1742540, 0xc000492780, 0x1742340, 0xc0004c3120, 0x1742540, 0xc0004c3100, 0xc000683080, 0xc00015a0a0, 0xc00015a0a0, 0x0)
	/Users/jasonhall/go/pkg/mod/github.com/google/go-containerregistry@v0.1.4-0.20201019153226-77421d5c5906/pkg/v1/mutate/rebase.go:53 +0x38b
github.com/buildpacks/imgutil/remote.(*Image).Rebase(0xc000094190, 0xc00056b040, 0x47, 0x1744100, 0xc00015a0a0, 0x1, 0x1)
	/Users/jasonhall/go/src/github.com/buildpack/imgutil/remote/remote.go:238 +0xd8
github.com/buildpacks/imgutil/remote_test.testImage.func16.1.2()
	/Users/jasonhall/go/src/github.com/buildpack/imgutil/remote/remote_test.go:727 +0x3e2
github.com/sclevine/spec.run(0xc000683080, 0xc000489db8, 0x1, 0x1)
	/Users/jasonhall/go/pkg/mod/github.com/sclevine/spec@v1.4.0/spec.go:300 +0x44
github.com/sclevine/spec.specHooks.run(0xc000150880, 0xc000150900, 0xc000683080, 0xc000150940)
	/Users/jasonhall/go/pkg/mod/github.com/sclevine/spec@v1.4.0/spec.go:281 +0x10c
github.com/sclevine/spec.Run.func3(0xc000683080, 0xc000309900, 0x3, 0x4, 0xc00033f1e0, 0x3, 0x4, 0x5f8e4c0a, 0x1, 0x1, ...)
	/Users/jasonhall/go/pkg/mod/github.com/sclevine/spec@v1.4.0/spec.go:257 +0x2d6
github.com/sclevine/spec.node.run.func1(0xc000683080)
	/Users/jasonhall/go/pkg/mod/github.com/sclevine/spec@v1.4.0/parser.go:133 +0x82
testing.tRunner(0xc000683080, 0xc000579de0)
	/usr/local/go/src/testing/testing.go:1108 +0xef
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:1159 +0x386
FAIL	github.com/buildpacks/imgutil/remote	16.481s
FAIL

shell returned 1
```

I modified remote's `subImage` to implement `ConfigFile()` by passing through to the underlying image, and tests passed. I admit I'm not entirely sure why, or whether this is not okay for some reason.

File: remote/remote.go [hunk 1]
```diff
@@ -17,8 +18,6 @@ import (
 	"github.com/google/go-containerregistry/pkg/v1/tarball"
 	"github.com/google/go-containerregistry/pkg/v1/types"
 	"github.com/pkg/errors"
-
-	"github.com/buildpacks/imgutil"
```
Comment (ekcasey) [inline remote/remote.go:21]: We group internal imports separately from external imports. Running `make format` will update this. We should really update the PR checks to fail on formatting inconsistencies.

**Review by ekcasey — CHANGES_REQUESTED** [review]
Thanks for the contribution :)

> I modified remote's subImage to implement ConfigFile() by passing through to the underlying image, and tests passed. I admit I'm not entirely sure why, or whether this is not okay for some reason.

I think the failure you saw is unrelated to you change and was instead caused by a combination of some weird code choices in `imgutil`'s remote rebase along with ggcr [implementation changes](https://github.com/google/go-containerregistry/commit/92c877ee115a51bd6e06937bd681103f20394843). I think the minimal fix you added if fine for now. We should refactor/clean-up this part of the code regardless.

**Comment by ImJasonH** [comment]
> I think the failure you saw is unrelated to you change and was instead caused by a combination of some weird code choices in `imgutil`'s remote rebase along with ggcr [implementation changes](https://github.com/google/go-containerregistry/commit/92c877ee115a51bd6e06937bd681103f20394843). I think the minimal fix you added if fine for now. We should refactor/clean-up this part of the code regardless.

Ah yeah that makes sense. Let me know if you'd like me to add a comment or TODO.

**Review by ekcasey — APPROVED** [review]

**Pull request merged**